huaweicloud/services/dew/resource_huaweicloud_csms_secret.go

```diff
@@ -257,16 +292,44 @@ func resourceCsmsSecretUpdate(ctx context.Context, d *schema.ResourceData, meta
 	}
 
 	// Update secret text
-	if d.HasChanges("secret_text") {
+	if d.HasChanges("secret_text", "secret_binary") {
 		opts := secrets.CreateVersionOpts{
-			SecretString: d.Get("secret_text").(string),
+			ExpireTime: d.Get("expire_time").(int),
+		}
+		if v, ok := d.GetOk("secret_text"); ok {
+			opts.SecretString = v.(string)
+		} else {
+			opts.SecretBinary = d.Get("secret_binary").(string)
```

Comment:
There is no need to add judgment, just splice it directly into the struct.

Same to create body